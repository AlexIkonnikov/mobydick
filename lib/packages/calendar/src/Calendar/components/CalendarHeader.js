"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const mobydick_core_1 = require("@lad-tech/mobydick-core");
const constants_1 = require("../constants");
const CalendarHeader = props => {
    const { title, onPressMonth, onPressYear, onPressLeft, onPressRight } = props;
    const [styles] = (0, mobydick_core_1.useStyles)(stylesCreate);
    return (<mobydick_core_1.View style={styles.container}>
      <mobydick_core_1.Pressable onPress={onPressLeft} accessibilityLabel={constants_1.LABELS.calendarLeftArrow}>
        <mobydick_core_1.SimpleIcon name={'icon-arrow-left'}/>
      </mobydick_core_1.Pressable>
      <mobydick_core_1.View style={styles.title}>
        {Boolean(title.currMonth) && (<mobydick_core_1.TouchableOpacity onPress={onPressMonth} accessibilityLabel={constants_1.LABELS.calendarPressTitleMonth}>
            <mobydick_core_1.Typography font={'Medium-Primary-M'}>{title.currMonth}</mobydick_core_1.Typography>
          </mobydick_core_1.TouchableOpacity>)}
        {Boolean(title.currYear) && (<mobydick_core_1.TouchableOpacity onPress={onPressYear} accessibilityLabel={constants_1.LABELS.calendarPressTitleYear}>
            <mobydick_core_1.Typography font={'Medium-Primary-M'}>{title.currYear}</mobydick_core_1.Typography>
          </mobydick_core_1.TouchableOpacity>)}
      </mobydick_core_1.View>

      <mobydick_core_1.Pressable onPress={onPressRight} accessibilityLabel={constants_1.LABELS.calendarRightArrow}>
        <mobydick_core_1.SimpleIcon name={'icon-arrow-right'}/>
      </mobydick_core_1.Pressable>
    </mobydick_core_1.View>);
};
exports.default = CalendarHeader;
const stylesCreate = (0, mobydick_core_1.createStyles)(_ => ({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        paddingTop: (0, mobydick_core_1.px)(12),
    },
    title: {
        width: (0, mobydick_core_1.px)(160),
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
    },
}));
