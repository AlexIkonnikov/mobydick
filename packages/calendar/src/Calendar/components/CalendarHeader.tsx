import {
  Pressable,
  rem,
  SimpleIcon,
  TouchableOpacity,
  Typography,
  useStyles,
  View,
} from '@npm/mobydick-core';
import React, {FC} from 'react';
import {StyleSheet} from 'react-native';

import {LABELS} from '../constants';

interface ICustomHeaderTitle {
  title: string;
  onPress: () => void;
  onPressLeft: () => void;
  onPressRight: () => void;
}

const CalendarHeader: FC<ICustomHeaderTitle> = props => {
  const {title, onPress, onPressLeft, onPressRight} = props;
  const [styles] = useStyles(stylesCreate);

  return (
    <View style={styles.container}>
      <Pressable
        onPress={onPressLeft}
        accessibilityLabel={LABELS.calendarLeftArrow}>
        <SimpleIcon name={'icon-arrow-left'} />
      </Pressable>
      <TouchableOpacity
        onPress={onPress}
        style={styles.title}
        accessibilityLabel={LABELS.calendarPressTitle}>
        <Typography>{title}</Typography>
      </TouchableOpacity>
      <Pressable
        onPress={onPressRight}
        accessibilityLabel={LABELS.calendarRightArrow}>
        <SimpleIcon name={'icon-arrow-right'} />
      </Pressable>
    </View>
  );
};

export default CalendarHeader;

const stylesCreate = () =>
  StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      paddingTop: rem(12),
    },
    title: {
      width: rem(160),
      alignItems: 'center',
    },
  });
