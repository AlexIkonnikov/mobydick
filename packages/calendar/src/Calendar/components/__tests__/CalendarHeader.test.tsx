import {render} from '@testing-library/react-native';
import React from 'react';

import CalendarHeader from '../CalendarHeader';

describe('CalendarHeader', () => {
  it('render', () => {
    const onPressMonth = jest.fn();
    const onPressYear = jest.fn();
    const onPressRight = jest.fn();
    const onPressLeft = jest.fn();

    const {toJSON} = render(
      <CalendarHeader
        title={{
          currMonth: 'currMonth',
          currYear: 'currentYear',
        }}
        onPressMonth={onPressMonth}
        onPressYear={onPressYear}
        onPressRight={onPressRight}
        onPressLeft={onPressLeft}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
