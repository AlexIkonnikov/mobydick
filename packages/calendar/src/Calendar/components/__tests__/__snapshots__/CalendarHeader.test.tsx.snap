// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarHeader render 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingTop": 36,
    }
  }
>
  <View
    accessibilityLabel="calendarLeftArrow"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      allowFontScaling={true}
      style={
        {
          "color": "#5E6678",
          "fontFamily": "Neotis",
          "fontSize": 72,
        }
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityLabel="calendarPressTitle"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "opacity": 1,
        "width": 480,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 45,
            "lineHeight": 57,
          },
          undefined,
        ]
      }
    >
      CalendarHeader
    </Text>
  </View>
  <View
    accessibilityLabel="calendarRightArrow"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      allowFontScaling={true}
      style={
        {
          "color": "#5E6678",
          "fontFamily": "Neotis",
          "fontSize": 72,
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
